### DISCUSSION TITLE
Regarding non conjugate priors

### DISCUSSION THREAD

**Discussion by prantik1998** [ReactiveBayes/RxInfer.jl#210] [Q&A]
I want to know how to perform inference with non conjugate priors using rxinfer

**Comment by albertpod** [comment]
Hi @prantik1998. Generally, two ways are available:

1. If you are in the realm of exponential family distributions, you can use CVI
```julia

f(x) = x
...
x ~ Beta(a, b)
x_ ~ f(x) where {meta = CVI(rng, nr_samples, nr_iterations, optimizer)}
y ~ Normal(0.0, x_)
...
```
More on it [here](https://biaslab.github.io/RxInfer.jl/stable/examples/advanced_examples/Conjugate-Computational%20Variational%20Message%20Passing/): 

2. Alternatively, you can define your own (efficient) [product](https://biaslab.github.io/BayesBase.jl/stable/#Base.prod-Tuple{BayesBase.UnspecifiedProd,%20Any,%20Any}) rule between the distributions you are going to multiply.

**Comment by prantik1998** [comment]
Thanks @albertpod 

I wanted to know whether the CVI method will also work when x is a vector ??

**Comment by Nimrais** [comment]
Hi @prantik1998, yes it will work. You can check this example https://biaslab.github.io/RxInfer.jl/stable/examples/advanced_examples/Nonlinear%20Sensor%20Fusion/#examples-nonlinear-sensor-fusion.
